src/BikeSharing.DomainLogic/BikeWorkout.cs

```diff
@@ -0,0 +1,328 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using BikeSharing.DomainLogic;
+
+namespace Training
+{
+
+    public interface IWorkout
+    {
+        DateTime Date { get; }
+        TimeSpan Duration { get; }
+        IHeartRate HeartRate { get; }
+        string Notes { get; set; }
+    }
+
+    public interface IHeartRate
+    {
+        double GetHeartRate();
+    }
+
+    public interface IDistanceWorkout : IWorkout
+    {
+        double Distance { get; }
+        double Pace { get; }
+    }
+
+    public interface IWorkoutType
+    {
+        WorkoutType Type { get; }
+    }
+
+    public interface IBikeWorkout : IDistanceWorkout, IWorkout
```

Comment:
agreed.  waht does IBikeWorkout *add* that's interesting?  also, IDistanceWorkout is an IWorkout, so you don' tneed to list both.